Nodejs/Product/Nodejs/SharedProject/FileChangeManager.cs

```diff
@@ -116,90 +171,83 @@ public void Dispose()
             this.disposed = true;
 
             // Unsubscribe from the observed source files.
-            foreach (var info in this.observedItems.Values)
+            foreach (var (ItemID, FileChangeCookie) in this.observedFiles.Values)
             {
-                ErrorHandler.ThrowOnFailure(this.fileChangeService.UnadviseFileChange(info.FileChangeCookie));
+                var hr = this.fileChangeService.UnadviseFileChange(FileChangeCookie);
+                if (ErrorHandler.Failed(hr)) { break; }
+                // don't want to crash VS during cleanup
+                Debug.Assert(ErrorHandler.Succeeded(hr), "UnadviseFileChange failed");
```

Comment:
>Debug.Assert(ErrorHandler.Succeeded(hr), "UnadviseFileChange failed"); [](start = 16, length = 70)

Aren't succeeded and failed mutually exclusive?

Reply:
yeah, the debug should be before the `break`